lib/after.js

```diff
@@ -35,27 +35,38 @@ test.afterEach( function() {
 	const locale = driverManager.currentLocale().toUpperCase();
 
 	if ( this.currentTest.state === 'failed' ) {
-		const prefix = `FAILED-${locale}-${screenSize}-${shortTestFileName}`;
-
 		let neverSaveScreenshots = config.get( 'neverSaveScreenshots' );
 		if ( neverSaveScreenshots ) {
 			return null;
 		}
 
-		driver.getCurrentUrl().then( ( url ) => console.log( `FAILED: Taking screenshot of: '${url}'` ), ( err ) => {
+		driver.getCurrentUrl().then( url => console.log( `FAILED: Taking screenshot of: '${url}'` ), ( err ) => {
 			slackNotifier.warn( `Could not capture the URL when taking a screenshot: '${err}'` );
 		} );
 
+
 		return driver.takeScreenshot().then( ( data ) => {
-			mediaHelper.writeScreenshot( data, prefix );
+			driver.getCurrentUrl().then( url => {
+				mediaHelper.writeScreenshot( data, function( date ) {
+					return `FAILED-${locale}-${screenSize}-${shortTestFileName}-${date}`;
+				}, {
+					url
+				} );
+			} );
 		}, ( err ) => {
 			slackNotifier.warn( `Could not take screenshot due to error: '${err}'` );
 		} );
 	}
+
 	if ( config.get( 'saveAllScreenshots' ) === true ) {
-		const prefix = [ `${locale}-${screenSize}-${shortDescribeFileName}`, `${shortTestFileName}` ];
 		return driver.takeScreenshot().then( ( data ) => {
-			mediaHelper.writeScreenshot( data, prefix, true );
+			driver.getCurrentUrl().then( url => {
+				mediaHelper.writeScreenshot( data, function( date ) {
+					return `${locale}-${screenSize}-${shortDescribeFileName}-${date}-${shortTestFileName}`;
```

Comment:
If `${date}` is used in both file names (here & on line 51) maybe it worth to put it right after other "static" values, so it would be easier to sort them etc?